frappe/automation/doctype/assignment_rule/assignment_rule.js

```diff
@@ -2,15 +2,74 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on('Assignment Rule', {
+	setup: function(frm) {
+		frm.trigger('setup_assignment_days_buttons');
+	},
+
 	refresh: function(frm) {
 		// refresh description
 		frm.events.rule(frm);
+		frm.trigger('set_field_options');
+	},
+
+	document_type: function(frm) {
+		frm.trigger('set_field_options');
 	},
+
+	setup_assignment_days_buttons: function(frm) {
+		const labels = ['Weekends', 'Weekdays', 'All Days'];
+		let get_days = (label) => {
+			const weekdays = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday'];
+			const weekends = ['Saturday', 'Sunday'];
+			return {
+				'All Days': weekdays.concat(weekends),
+				'Weekdays': weekdays,
+				'Weekends': weekends,
+			}[label];
+		};
+
+		let get_button_html = (label) => `<button
```

Comment:
This is cool UX.

Recommend to create a `Button` element in the framework and separate (and standardize) the implementation details so that does not become a specific.

In general, in a controller, we should try and use configuration over direct HTML so that when we do redesigns (or there is another UI layer or view), this will be come much easier.